internal/dispatch/graph/lookupsubjects_test.go

```diff
@@ -707,6 +708,215 @@ func TestCaveatedLookupSubjects(t *testing.T) {
+			},
+			ONR("document", "somedoc", "view"),
+			RR("user", "..."),
+			[]*v1.FoundSubject{
+				{
+					SubjectId: "fred",
+				},
+			},
+		},
+		{
+			"all arrow over multiple resource IDs",
+			`definition user {}
+	
+			 definition organization {
+			 	relation member: user
+			 }
+
+			 definition folder {
+   			    relation parent: organization
+				permission view = parent.all(member)
+			 }
+
+		 	 definition document {
+				relation folder: folder
+				permission view = folder->view
+  		 }`,
+			[]*corev1.RelationTuple{
+				tuple.MustParse("document:somedoc#folder@folder:folder1"),
+				tuple.MustParse("document:somedoc#folder@folder:folder2"),
+				tuple.MustParse("folder:folder1#parent@organization:org1"),
+				tuple.MustParse("folder:folder2#parent@organization:org2"),
+				tuple.MustParse("folder:folder2#parent@organization:org3"),
+				tuple.MustParse("organization:org1#member@user:fred"),
+				tuple.MustParse("organization:org2#member@user:tom"),
+				tuple.MustParse("organization:org3#member@user:tom"),
+				tuple.MustParse("organization:org2#member@user:sarah"),
+			},
+			ONR("document", "somedoc", "view"),
+			RR("user", "..."),
+			[]*v1.FoundSubject{
+				{
+					SubjectId: "tom",
+				},
+				{
+					SubjectId: "fred",
+				},
+			},
+		},
+		{
+			"intersection arrow over caveated teams",
+			`definition user {}
+	
+			 definition team {
+			 	relation member: user | user with anothercaveat
+			 }
+
+			 caveat caveat1(someparam1 int) {
+				someparam1 == 42
+			 }
+
+ 			 caveat caveat2(someparam2 int) {
+				someparam2 == 42
+			 }
+
+			 caveat anothercaveat(anotherparam int) {
+			 	anotherparam == 43
+			 }
+
+		 	 definition document {
+				relation team: team with caveat1 | team with caveat2
+				permission view = team.all(member)
+  		 }`,
+			[]*corev1.RelationTuple{
+				tuple.MustParse(`document:somedoc#team@team:team1[caveat1:{":someparam1":42}]`),
+				tuple.MustParse(`document:somedoc#team@team:team2[caveat2:{":someparam2":43}]`),
+				tuple.MustParse(`team:team1#member@user:tom`),
+				tuple.MustParse(`team:team2#member@user:tom`),
+				tuple.MustParse(`team:team1#member@user:fred`),
+				tuple.MustParse(`team:team2#member@user:fred[anothercaveat:{":anotherparam":43}]`),
+				tuple.MustParse(`team:team1#member@user:sarah`),
+			},
+			ONR("document", "somedoc", "view"),
+			RR("user", "..."),
+			[]*v1.FoundSubject{
+				{
+					SubjectId: "tom",
+					CaveatExpression: caveatAnd(
+						caveatAndCtx("caveat1", map[string]interface{}{"someparam1": 42}),
+						caveatAndCtx("caveat2", map[string]interface{}{"someparam2": 43}),
+					),
+				},
+				{
+					SubjectId: "fred",
+					CaveatExpression: caveatAnd(
+						caveatAnd(
+							caveatAndCtx("caveat1", map[string]interface{}{"someparam1": 42}),
+							caveatAndCtx("caveat2", map[string]interface{}{"someparam2": 43}),
+						),
+						caveatAndCtx("anothercaveat", map[string]interface{}{"anotherparam": 43}),
+					),
+				},
+			},
+		},
```

Comment:
I don't see any test cases for (not just for LR, the other methods too):

- an `all` over a recursive relationship
- an `all` that is part of an expression

Reply:
Added